microshift.sh

```diff
@@ -0,0 +1,134 @@
+if ! sudo subscription-manager status >& /dev/null ; then
+   echo "machine must be registered using subscription-manager"
+   exit 1
+fi
+
+run_preflight_checks ${BUNDLE_TYPE}
+rm -fr ${INSTALL_DIR} && mkdir ${INSTALL_DIR}
+
+sudo virsh destroy ${SNC_PRODUCT_NAME} || true
+sudo virsh undefine --nvram ${SNC_PRODUCT_NAME} || true
+sudo rm -fr /var/lib/libvirt/images/${SNC_PRODUCT_NAME}.qcow2
+sudo rm -fr /var/lib/libvirt/images/microshift-installer-*.iso
+
+
+# Generate a new ssh keypair for this cluster
+# Create a 521bit ECDSA Key
+rm id_ecdsa_crc* || true
+ssh-keygen -t ecdsa -b 521 -N "" -f id_ecdsa_crc -C "core"
+
+# This requirement is taken from https://github.com/openshift/microshift/blob/main/scripts/image-builder/configure.sh
+# Also https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/composing_a_customized_rhel_system_image/installing-composer_composing-a-customized-rhel-system-image
+# list out the dependencies and usecase.
+function configure_host {
+    sudo dnf install -y git osbuild-composer composer-cli ostree rpm-ostree \
+      cockpit-composer cockpit-machines bash-completion podman genisoimage \
+      createrepo yum-utils selinux-policy-devel jq wget lorax
+    sudo systemctl enable osbuild-composer.socket --now
+    sudo systemctl enable cockpit.socket --now
+    sudo firewall-cmd --add-service=cockpit
+    # The home dir and files must have read permissions to group
+    # and others because osbuilder is running from another non-priviledged user account
+    # and allow it to read the files on current user home, it is required.
+    # https://github.com/openshift/microshift/blob/main/scripts/image-builder/configure.sh#L29-L32
+    sudo chmod 0755 $HOME
+}
+
+function enable_repos {
+    sudo subscription-manager repos \
+       --enable rhocp-4.12-for-rhel-8-$(uname -i)-rpms \
+       --enable fast-datapath-for-rhel-8-$(uname -i)-rpms
+}
+
+function download_microshift_rpm {
+    local pkgDir=$1
+    pushd ${pkgDir}
+    brew download-build --quiet --arch noarch --arch ${ARCH} ${MICROSHIFT_NVR}
+    popd
+}
+
+function create_iso {
+    local pkgDir=$1
+    rm -fr microshift
+    git clone -b release-4.12 https://github.com/openshift/microshift.git
+    pushd microshift
+    sed -i '/# customizations/,$d' scripts/image-builder/config/blueprint_v0.0.1.toml
+    cat << EOF >> scripts/image-builder/config/blueprint_v0.0.1.toml
+[[packages]]
+name = "microshift-release-info"
+version = "*"
+EOF
+    sed -i 's/redhat/core/g' scripts/image-builder/config/kickstart.ks.template
+    sed -i '/--bootproto=dhcp/a\network  --hostname=api.crc.testing' scripts/image-builder/config/kickstart.ks.template
+    scripts/image-builder/cleanup.sh -full
+    scripts/image-builder/build.sh -microshift_rpms ${pkgDir} -pull_secret_file ${OPENSHIFT_PULL_SECRET_PATH} -authorized_keys_file $(realpath ../id_ecdsa_crc.pub)
+    popd
+}
+
+enable_repos
+configure_host
+microshift_pkg_dir=$(mktemp -p /tmp -d tmp-rpmXXX)
```

Comment:
No since we are not using `build.sh` so `_output/rpmbuild` directory is not created and this is just downloading the rpm file doesn't matter the directory since we remove it at the end.

Reply:
`make rpm` not used.

Reply:
This does not have much to do with `make rpm`, `mktemp` becomes `mkdir -p _output/rpmbuild`, and then we can skip the `-microshift_rpms ${pkgDir}` argument to `build.sh`, we don't need the `chmod` nor to delete it afterwards.